PF/PF-Assignment2/ShoppingCart.java

```diff
@@ -0,0 +1,225 @@
+			return this.code;
+		}
+		
+		/*
+		 * @param   nothing
+		 * @return  start date 
+		 */
+		public String getStartDate() {
+			return this.startDate;
+		}
+		
+		/*
+		 * @param   nothing
+		 * @return  end date
+		 */
+		public String getEndDate() {
+			return this.endDate;
+		}
+		
+		private PromotionEnum(String code, String startDate, String endDate) {
+			this.code = code;
+			this.startDate = startDate;
+			this.endDate = endDate;
+		}
+	}
+	
+	static class Item{
+		private long itemID;
+		private String itemName;
+		private double price;
+		private long quantity;
+		private Item next;
+		
+		Item(long itemID, String itemName, double price, long quantity){
+			this.itemID = itemID;
+			this.itemName = itemName;
+			this.price = price;
+			this.quantity = quantity;
+			this.next = null;
+		}
+	}
+	
+	/*
+	 * @param   item id (int)
+	 * @return  item price
+	 */
+	public double getPrice(long itemID) {
+		Item item;
+		item = this.searchItem(itemID);
+		return item.price;
+	}
+	
+	/*
+	 * @param   item id (int), price of item (double)
+	 * @return  nothing
+	 * Update Item price after discount
+	 */
+	public void setPrice(long itemID, double price) {
+		Item item;
+		item = this.searchItem(itemID);
+		item.price = price;
+	}
+	
+	/*
+	 * @param   nothing
+	 * @return  true if Shopping Cart is Empty otherwise false
+	 */
+	public boolean isEmpty() {
+		if (this.firstItem == null)
+			return true;
+		return false;
+	}
+	
+	/*
+	 * @param   item name (string), item price (double), item quantity(long)
+	 * @return  nothing
+	 * Add item into the Shopping Cart list
+	 */
+	public void add(long itemID, String itemName, double price, long quantity) {
+		Item newItem = new Item(itemID, itemName, price, quantity);
+		total += price * (double)quantity;
+		if(isEmpty()) {
+			firstItem = newItem;
+			lastItem = newItem;
+		}
+		else {
+			lastItem.next = newItem;
+			lastItem = lastItem.next;
+		}
+	}
+	
+	/*
+	 * @param   Item id 
+	 * @return  nothing
+	 * Remove Item from list of Shopping cart
+	 */
+	public void remove(long itemID) {
+		Item it = firstItem;
+		Item pre = firstItem;
+		while(it != null) {
+			if(it.itemID == itemID)
+				break;
+			pre = it;
+			it = it.next;
+		}
+		if(it == firstItem) 
+			firstItem = firstItem.next;
+		else if(it == lastItem) {
+			lastItem = pre;
+			pre.next = null;
+		}
+		else 
+			pre.next = it.next;
+		System.out.println("Item id-" + it.itemID + " Removed !");
+		it = null;
+	}
+	
+	/*
+	 * @param   item-id (long)
+	 * @return  Item reference if it is found otherwise null
+	 */
+	public Item searchItem(long item_id) {
+		Item it = firstItem;
+		while(it != null) {
+			if(it.itemID == item_id)
+				return it;
+			it = it.next;
+		}
+		return null;
+	}
+	
+	/*
+	 * @param   item-id(long), change ("+" or "-")(String), Quantity(long)
+	 * @return  nothing
+	 * + used for Increase Quantity
+	 * - used for Decrease Quantity
+	 */
+	public void changeQuantity(long itemID, String change, long quantity) {
```

Comment:
Can be simplified like : just take quantity changed not by which quantity is being changed.